apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/dao/state/poll/VoteTable.java

```diff
@@ -0,0 +1,124 @@
+package com.apollocurrency.aplwallet.apl.core.dao.state.poll;
+
+import com.apollocurrency.aplwallet.apl.core.app.Blockchain;
+import com.apollocurrency.aplwallet.apl.core.app.CollectionUtil;
+import com.apollocurrency.aplwallet.apl.core.app.Transaction;
+import com.apollocurrency.aplwallet.apl.core.app.Vote;
+import com.apollocurrency.aplwallet.apl.core.db.DbClause;
+import com.apollocurrency.aplwallet.apl.core.db.DbIterator;
+import com.apollocurrency.aplwallet.apl.core.db.DbKey;
+import com.apollocurrency.aplwallet.apl.core.db.LongKeyFactory;
+import com.apollocurrency.aplwallet.apl.core.db.derived.EntityDbTable;
+import com.apollocurrency.aplwallet.apl.core.entity.state.poll.Poll;
+import com.apollocurrency.aplwallet.apl.core.transaction.messages.MessagingVoteCasting;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.enterprise.inject.spi.CDI;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+
+@Slf4j
+@Singleton
+public class VoteTable extends EntityDbTable<Vote> {
+    private static final LongKeyFactory<Vote> VOTE_LONG_KEY_FACTORY = new LongKeyFactory<>("id") {
+        @Override
+        public DbKey newKey(Vote vote) {
+            return vote.getDbKey();
+        }
+    };
+    private PollTable pollTable;
+
+    @Inject
+    public VoteTable(PollTable pollTable) {
+        super("vote", VOTE_LONG_KEY_FACTORY);
+        this.pollTable = pollTable;
+    }
+
+
+    @Override
+    public void save(Connection con, Vote entity) throws SQLException {
+        Blockchain blockchain = CDI.current().select(Blockchain.class).get();
```

Comment:
Incorrect implementation. see as example 'CurrencyTransferTable'.
The height should be assigned outside 'Table', usually inside 'service' class before storing entity.
`...CDI.current().select(Blockchain.class).get()` should NOT be used here.

In Service you can use:  BlockChainInfoService blockChainInfoService;  for getting current height.